src/Readers/ColumnIndexReader.cs

```diff
@@ -30,7 +30,7 @@ public ColumnIndexValueReader(int columnIndex)
 
         public bool TryGetValue(ExcelSheet sheet, int rowIndex, IExcelDataReader reader, out ReadCellValueResult result)
         {
-            if (ColumnIndex > reader.ResultsCount)
+            if (ColumnIndex >= reader.FieldCount)
```

Comment:
failed with `IndexOutOfRangeException` for the boundary value of `ColumnIndex == reader.FieldCount`